controllers/virtualmachine/virtualmachine/virtualmachine_controller.go

```diff
@@ -451,9 +453,15 @@ func (r *Reconciler) ReconcileNormal(ctx *pkgctx.VirtualMachineContext) (reterr
 		r.Recorder.EmitEvent(ctx.VM, "ReconcileNormal", err, true)
 	}
 
-	// Always check if this VM needs to be in the probe queue. We want to do this even if
-	// the VM hasn't been created yet so the condition is updated.
-	r.Prober.AddToProberManager(ctx.VM)
+	if !pkgcfg.FromContext(ctx).Features.WorkloadDomainIsolation ||
+		pkgcfg.FromContext(ctx).AsyncSignalDisabled {
+
+		// Add the VM to the probe manager. This is idempotent.
+		r.Prober.AddToProberManager(ctx.VM)
+	} else if p := ctx.VM.Spec.ReadinessProbe; p != nil && p.TCPSocket != nil {
+		// TCP probes still use the probe manager.
+		r.Prober.AddToProberManager(ctx.VM)
+	}
```

Comment:
I think we may need an `else { r.Prober.RemoveFromProberManager(ctx.VM) } ` here for if the probe is removed. I think the existing code is mostly a noop when the worker finds the `Spec.ReadinessProbe` empty but I don't think anything else will remove it from the vmReadinessProbeList until the VM is deleted.

Reply:
Ah, in case the probe type changes, got it.